### PR TITLE
Forcefield Projector Balance

### PR DESCRIPTION
**Pull request by thatoneplebeian**
## About The Pull Request

Adds a use time to forcefield projectors (set at 1 second for now, easy to change) and makes EMP blasts destroy the forcefields. Multiple forcefields cannot be deployed at once.
Separates the charge of the projector from the shield's integrity; charge is only drained by having active shields and not by hitting the shields.

## Why It's Good For The Game

Forcefield projectors are kind of overpowered, they can easily make an instant wall with quite a lot of health. The use time makes it much harder to abuse. EMP blasts destroying forcefields not only helps make this a more balanced item but is consistent with EMPs disrupting other 'projection' items, like the holodeck.

## Changelog
:cl:
balance: Forcefield projectors now take time to use, and their forcefields can be destroyed with EMPs.
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
I agree that forcefield projectors are a bit busted, but what purpose does it serve in this state? Would it even see any use?

File: code/game/objects/items/devices/forcefieldprojector.dm [hunk 1]
```diff
@@ -16,6 +16,8 @@
 	var/max_fields = 3
 	var/list/current_fields
 	var/field_distance_limit = 7
+	var/forcefield_busy = FALSE //Prevents spamming forcefields
+	var/creation_time = 15 //Time needed to create, in deciseconds
```
Comment (Jared-Fogle) [inline code/game/objects/items/devices/forcefieldprojector.dm:20]: These are doc comments, so use `///` instead.

Comment (Jared-Fogle) [reply code/game/objects/items/devices/forcefieldprojector.dm:20]: Also `creation_time` should probably just be `1.5 SECONDS`.

**Comment by halitosisman** [comment]
No one uses forcefields outside of Xenobio.

**Comment by cacogen** [comment]
What purpose does it serve to begin with? It seems like a generic/prototypical version of the specialised projectors multiple departments have.

**Comment by trerri** [comment]
Delete the projector, it's just a departmental projector but the department is greytide

**Comment by Krysonism** [comment]
Have they not already been nerfed?

They couldn't keep the doctor + borg out who wanted to forcefully amputate my tail.

**Comment by cacogen** [comment]
Was 50 more damage than a person not enough time to escape? Guess it was fate

**Comment by halitosisman** [comment]
Were you by any chance the nukie that I killed with a forcefield projector once?

**Comment by Angustmeta** [comment]
> Have they not already been nerfed?
> 
> They couldn't keep the doctor + borg out who wanted to forcefully amputate my tail.

How these currently work is a bit unintuitive. All forcefields from same projector share HP, and that HP = %charge of the device. Charge drains from having active fields and when damage is dealt to them, and regens when there's no fields. 

So if the device is used when near 0% charge, the fields are destroyed near instantly.

**Comment by thatoneplebeian** [comment]
> 
> So uhh, can you explain a single useful purpose that FPs will have after this PR?

They would serve the exact same purpose, I haven't modified their core function, which is to block passage through the turf. 

I know what you mean, though.
Admittedly, the 1.5 seconds is a little long. It might be better if I adjust it to 1 second and make it possible to start putting up multiple shields at once. I am only able to test if the code functions, I don't have anyone to actually playtest the code with and tweak the balance accordingly.

As @cacogen brought up, the issue #52933 covers why the forcefield projector is overpowered, and this pull request addresses some of those complaints. 

If the forcefield projector would never be used once nerfed no matter how small the nerf is, then it definitely would just be best to delete it outright, but I don't think that's the case. 
If enough people dislike the use time that much I could address one of the other factors that make the forcefield op. EMP blast seems like it should have been in there from the get-go and I would rather not change that.

**Comment by SkeletalElite** [comment]
Is the integrity of the barrier even used? I was under the impression the only thing that mattered was the charge of the projector and that hitting the fields reduced its charge

**Comment by thatoneplebeian** [comment]
The projector now acts in a more intuitive manner. Hitting fields does not drain the charge of the projector. Projecting a field while the charge is low still gives it only a short time to live, but that time is not shortened by someone hitting the shield.

**Comment by Axanthic** [comment]
Wow good idea lets take something no one uses and insure no one even thinks about using it, whoa you trapped someone in a circle because they didn't move for 4 seconds straight, not to mention they were next to something which was a requirement for you to trap them and that you have to stay within eyesight otherwise the fields break? Oh and they'll break in 30 seconds or 10 if the person just wacks them.

Thank god, this is like wow, honestly the most important thing ss13 needs, I think we can officially refer to TG as ss14 going forward.

**Comment by ExcessiveUseOfCobblestone** [comment]
why the new integrity var instead of using preexisting obj integrity structure?

**Comment by thatoneplebeian** [comment]
I wasn't aware such a thing existed. I can update it to use already existing code.

**Comment by Doosteroni** [comment]
The problem with the Forcefield projector is its only purpose currently is to validhunt. Just fucking remove it

**Comment by ExcessiveUseOfCobblestone** [comment]
Use the pre-existing obj_integrity system and Repr

**Pull request closed without merging**